src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -92,20 +98,50 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
                             }
 
                             builder.Add(new BoundStringInsert(interpolation, value, alignment, format, null));
+                            if (value.ConstantValue != null)
+                            {
+                                if (!value.ConstantValue.IsString || value.ConstantValue.IsBad || value.ConstantValue.IsNull)
```

Comment:
Right now the logic to determine if this value is constant is spread out to four places: the format check, the alignment check, the constant check and the type of constant. Consider just combining it into a single check here:

```c#
if (value.ConstantValue?.IsString == true &&
    !value.ConstantValue.IsBad &&
    !value.ConstantValue.IsNull &&
    interpolation.FormatClaues is null && 
    interpolation.AlignmentClause is null) 
{
   ...
}
```

Reply:
The first 3 can be collapsed to a single pattern: `value.ConstantValue is { IsString: true, IsBad: false, IsNull: false }`

---
In reply to: [447268625](https://github.com/dotnet/roslyn/pull/45396#discussion_r447268625) [](ancestors = 447268625)

Reply:
And the second 2 as well: `interpolation is { FormatClauses : null, AlignmentClause: null }`

---
In reply to: [447343672](https://github.com/dotnet/roslyn/pull/45396#discussion_r447343672) [](ancestors = 447343672,447268625)

Reply:
We can also bail out early if `!constantEnabled`.

---
In reply to: [447343824](https://github.com/dotnet/roslyn/pull/45396#discussion_r447343824) [](ancestors = 447343824,447343672,447268625)

Reply:
Thanks for all the suggestions, I've incorporated them into commit 16.

---
In reply to: [447347863](https://github.com/dotnet/roslyn/pull/45396#discussion_r447347863) [](ancestors = 447347863,447343824,447343672,447268625)